### PR TITLE
[luci] Revise RemoveRedundantTransposePass

### PR DESCRIPTION
**Pull request by llFreetimell**
Parent Issue : #5777

Current `RemoveRedundantTransposePass` has following problems

- The pass is ended right after the pass is applied to only one node.
- Checking `CircleTranspose` is done twice.

As a result, if there are a lot of nodes to apply this pass, much time is needed.
To make this pass faster, this commit will revise following things.

- Do not break right after this pass is applied for a node. Apply this pass as much as possible at once.
- Filtering `CircleTranspose` first and then execute internal function.

ONE-DCO-1.0-Signed-off-by: Seok NamKoong <seok9311@naver.com>

### CODE REVIEW

File: compiler/luci/pass/src/RemoveRedundantTransposePass.cpp [hunk 1]
```diff
@@ -118,11 +115,10 @@ bool RemoveRedundantTransposePass::run(loco::Graph *g)
   bool changed = false;
   for (auto node : loco::active_nodes(loco::output_nodes(g)))
   {
-    auto circle_node = loco::must_cast<luci::CircleNode *>(node);
-    if (remove_consecutive_transpose_function(circle_node))
+    if (auto transpose = dynamic_cast<luci::CircleTranspose *>(node))
     {
-      changed = true;
-      break;
```
Comment (seanshpark) [inline compiler/luci/pass/src/RemoveRedundantTransposePass.cpp:125]: @struss , did you have any problem to add a `break` here?

Comment (struss) [reply compiler/luci/pass/src/RemoveRedundantTransposePass.cpp:125]: Yes, but I think that `break` make this pass only update once for `RedundantTranspose` case. I think removing this nice.

Comment (seanshpark) [reply compiler/luci/pass/src/RemoveRedundantTransposePass.cpp:125]: > Yes, 

What problem did you have?

Comment (struss) [reply compiler/luci/pass/src/RemoveRedundantTransposePass.cpp:125]: > > Yes,
> 
> What problem did you have?

Sorry to miss reply. I think removing that break make no problem on it. I first thought only one change would okay, so I added. but think about more deeply, this change could be done multiple. so I think removing that `break` is nice change.

> @struss , did you have any problem to add a `break` here?

Sorry for confusing. to reply on yes. I have no problem on this. Sorry.

**Review by seanshpark — APPROVED** [review]
LGTM

**Review by struss — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**